To-do list/Controllers/TaskDetailsViewController.swift

```diff
@@ -7,91 +7,42 @@
 
 import UIKit
 
+protocol TaskDetailsViewControllerDelegate: AnyObject {
+    func taskDetails(_ controller: TaskDetailsViewController, didCreateUpdate task: Task)
+}
+
 class TaskDetailsViewController: UIViewController {
     
+    weak var delegate: TaskDetailsViewControllerDelegate?
+    
     @IBOutlet weak var titleTextField: UITextField!
     @IBOutlet weak var descriptionTextField: UITextField!
     @IBOutlet weak var statusButtonOutlet: UIButton!
+  
+    enum ViewType {
+        case createTask
+        case updateTask
+    }
     
-    var users = [User]()
-    var userIndex = Int()
-    var taskIndex = Int()
-    var counter = 0
-    var status = "to do"
-    var titleField = ""
-    var choosenTableViewCell = Bool()
+    var task = Task()
+    var viewType = ViewType.createTask
+    var id = Int()
     
     override func viewDidLoad() {
         super.viewDidLoad()
         
-        users = UserDefaultsManager.shared.getValueForUser() ?? [User]()
-                titleTextField.text = titleField
-        if choosenTableViewCell {
-            titleTextField.text = users[userIndex].task?[taskIndex].title
-            descriptionTextField.text = users[userIndex].task?[taskIndex].description
-            
-            switch users[userIndex].task?[taskIndex].status {
-            case "in progress":
-                statusButtonOutlet.backgroundColor = .systemOrange
-                statusButtonOutlet.setTitle("in progress", for: .normal)
-            case "done":
-                statusButtonOutlet.backgroundColor = .systemRed
-                statusButtonOutlet.setTitle("done", for: .normal)
-            case "to do":
-                statusButtonOutlet.backgroundColor = .systemGreen
-                statusButtonOutlet.setTitle("to do", for: .normal)
-            default:
-                break
-            }
+        if viewType == .updateTask {
+            titleTextField.text = task.title
+            descriptionTextField.text = task.description
+            id = task.id
+            StatusButton.updateStatus(statusButtonOutlet, task.status)
```

Comment:
move this code to a separate func like: func prepareUIElements()

Reply:
done